### DISCUSSION TITLE
Deployment Issues on Azure App Service

### DISCUSSION THREAD

**Discussion by qaiseralikhan** [strapi/strapi#6304] [Legacy Topics]
Hi there,

I am using Azure App Service (linux) to deploy the Strapi app. I have used GitHub deployment method and using Github Actions for pipeline. The pipeline is completed perfectly fine and the app successfully deploys to the App Service.   But the log shows the following error regarding sharp package:

![Error](https://user-images.githubusercontent.com/38536029/82494252-bc52bc00-9b02-11ea-8234-053c019cf4d3.png)

Here's the github action yml:

`name: Build and deploy Node.js app to Azure Web App - et-strapi-sb

on:
  push:
    branches:
      - master

jobs:
  build-and-deploy:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@master

    - name: Set up Node.js version
      uses: actions/setup-node@v1
      with:
        node-version: '12.x'

    - name: yarn install, build
      run: |
        yarn cache clean
        yarn
        yarn add sharp
        NODE_ENV=production yarn run build 
        
    - name: 'Deploy to Azure Web App'
      uses: azure/webapps-deploy@v1
      with:
        app-name: 'estr'
        slot-name: 'production'
        publish-profile: ${{ secrets.AzureAppService_PublishProfile_e4746ddc772945849790c5f5a4443738 }}
        package: .`

Please help me resolving this issue.

thanks

**Expected behavior**
App should be up & running.

**System**

- Node.js version: 12.16.3
- NPM version: 6
- Strapi version: 3.0.0@beta
- Database: sqlite
- Operating system: ubuntu

**Comment by derrickmehaffy** [comment]
you will need to install the `build-essential` and `libpng-dev` packages if the app service's linux is based on ubuntu.

**Comment by qaiseralikhan** [comment]
Hi, thanks for the reply. 
I tried installing these by accessing azure app service through SSH, but it didn't work. I also tried re-deploying using Github Actions workflow, same as before, but it didn't work either. It keeps throwing sharp module error *File too short*, as in the screenshot above. 
Is there any other solution you can suggest please?
Thanks.

**Comment by derrickmehaffy** [comment]
I do not have any other suggestions, but in this case I'm going to move this over to a discussion as it's not directly an issue with Strapi but the environment it's being installed in.

**Comment by ahrherrera** [comment]
Hi, I am facing a simlar problem but I did make build in Azure but does not run.

Here: https://stackoverflow.com/questions/62355445/strapi-on-azure-does-not-run
